scanpipe/models.py

```diff
@@ -633,6 +635,16 @@ def delete(self, *args, **kwargs):
 
         return super().delete(*args, **kwargs)
 
+    def mark_for_deletion(self):
+        self.is_marked_for_deletion = True
+        self.save()
```

Comment:
Using save() as-is may have unwanted consequences.
You should at least provide the update_fields value, or better, use the `self.update(is_marked_for_deletion=True)` method.